bottlecap/src/lifecycle/invocation/processor.rs

```diff
@@ -516,19 +526,20 @@ impl Processor {
             dropped_p0_spans: 0,
         };
 
-        let send_data: SendData = trace_processor
+        let send_data = trace_processor
             .process_traces(
                 self.config.clone(),
                 tags_provider.clone(),
                 header_tags,
                 vec![traces],
                 body_size,
                 self.inferrer.span_pointers.clone(),
+                api_key,
```

Comment:
Passing `api_key` to `process_traces()`, so `process_traces()` won't need to handle failure inside.